cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/class-media.php

```diff
@@ -122,16 +130,35 @@ public function __construct( Plugin $plugin ) {
 	public function is_media( $attachment ) {
 		$is_media = false;
 		if ( 'attachment' === get_post_type( $attachment ) ) {
-			$is_media = $this->get_post_meta( $attachment, 'is_media', true );
-			if ( '' === $is_media ) {
-				$is_media = wp_attachment_is( 'image', $attachment ) || wp_attachment_is( 'video', $attachment );
-				$this->update_post_meta( $attachment, 'is_media', $is_media );
-			}
+			/**
+			 * Filter the default Cloudinary Media Types.
+			 *
+			 * @param array $types The default media types array.
+			 *
+			 * @return array
+			 */
+			$media_types = apply_filters( 'cloudinary_media_types', array( 'image', 'video', 'audio' ) );
+			$type        = $this->get_media_type( $attachment );
+			$is_media    = in_array( $type, $media_types );
 		}
 
 		return $is_media;
 	}
 
+	/**
+	 * Get a resource type based on file.(Cloudinary v1 remove mime type in post data).
+	 *
+	 * @param \WP_Post|int $attachment_id The attachment ID or object.
```

Comment:
@pereirinha - Correct. I was expecting an object since `get_post_type` accepts both. But I've change it to be more consistent by having `attachment_id` for all methods. I'll change this to int.